app/modules/sntp.c

```diff
@@ -805,15 +809,23 @@ static int sntp_sync (lua_State *L)
       server_count++;
     }
   } else if (server_count == 0) {
-    // default to ntp pool
     lua_newtable(L);
-    int i;
-    for (i = 0; i < 4; i++) {
-      lua_pushnumber(L, i + 1);
-      char buf[64];
-      c_sprintf(buf, "%d.nodemcu.pool.ntp.org", i);
-      lua_pushstring(L, buf);
-      lua_settable(L, -3);
+    struct netif *iface = (struct netif *)eagle_lwip_getif(0x00);
+    ip_addr_t ntp_addr = iface->dhcp->offered_ntp_addr;
```

Comment:
Is this always guaranteed to be set? What happens if the interface was manually configured an didn't use dhcp?

Reply:
Is this a trick question?
The real answer is "I don't know and havn't tested this case". looks like it should be: 
```
ip_addr_t ntp_addr = iface->dhcp ? iface->dhcp->offered_ntp_addr : NULL;
```